src/js/angular/ttyg/directives/chat-item-detail.directive.js

```diff
@@ -50,13 +57,78 @@ function ChatItemDetailComponent(toastr, $translate, TTYGContextService, Markdow
                 $scope.onRegenerateQuestion({chatItem: $scope.chatItemDetail});
             };
 
-            $scope.copyAnswerToClipboard = () => {
-                $scope.onCopyAnswerToClipboard({chatItem: $scope.chatItemDetail});
+            /**
+             * Copies the message to the clipboard.
+             *
+             * @param {string} message
+             */
+            $scope.copy = (message) => {
+                TTYGContextService.emit(TTYGEventName.COPY_TEXT_TO_CLIPBOARD, message);
+            };
+
+            /**
+             * Extract the explanation of how the answer was generated.
+             */
+            $scope.explainResponse = () => {
+                const explainResponse = TTYGContextService.getExplainResponseCache()[$scope.chatItemDetail.answer.id];
```

Comment:
Don't leak implementation details! Just implement a method for getting this by id

Reply:
Refactored